spec/lib/my_api_client/request/basic_spec.rb

```diff
@@ -32,14 +28,44 @@ def _request_with_relative_uri(http_method, pathname, headers, query, body); end
 
       before { allow(instance).to receive(:_request_with_relative_uri).and_return(response) }
 
-      it 'calls the request method with relative URL' do
-        execute
-        expect(instance).to have_received(:_request_with_relative_uri)
-          .with(http_method, pathname, headers, query, body)
+      context 'when the block is not given' do
+        subject(:execute) do
+          instance.public_send(http_method, pathname, headers: headers, query: query, body: body)
+        end
+
+        it 'calls the request method with relative URL' do
+          execute
+          expect(instance).to have_received(:_request_with_relative_uri)
+            .with(http_method, pathname, headers, query, body)
+        end
+
+        it 'returns a response body object' do
+          expect(execute).to eq resource
+        end
       end
 
-      it 'returns a response body object' do
-        expect(execute).to eq resource
+      context 'when the block is given' do
+        subject(:execute) do
+          instance.public_send(
+            http_method,
+            pathname,
+            headers: headers,
+            query: query,
+            body: body
+          ) do |response|
+            response
+          end
+        end
+
+        it 'calls the request method with relative URL' do
+          execute
+          expect(instance).to have_received(:_request_with_relative_uri)
+            .with(http_method, pathname, headers, query, body)
+        end
+
+        it 'passes the sawyer response to the block parameter and returns the block result' do
+          expect(execute).to eq response
+        end
```

Comment:
It should be tested with `yield_with_args` matcher:

```rb
it do
  expect { |b| instance.public_send(..., &b) }.to yield_with_args(Sawyer::Response)
end
```

And should be explicit that the method returns a block result:

```rb
it do
  expect(instance.public_send(...) { 'a return value of block' })
    .to eq 'a return value of block'
end
```

Reply:
Fixed on cc3c41284040e9879ae8bd5e3beafbbb0e1f93a6


https://github.com/ryz310/my_api_client/blob/cc3c41284040e9879ae8bd5e3beafbbb0e1f93a6/spec/lib/my_api_client/request/basic_spec.rb#L66-L76